src/TestFramework/AdapterInstaller.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\TestFramework;
+
+use Composer\Autoload\ClassLoader;
+use Infection\FileSystem\Finder\ComposerExecutableFinder;
+use function Safe\sprintf;
+use Symfony\Component\Process\Process;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ */
+final class AdapterInstaller
+{
+    private const OFFICIAL_ADAPTERS = ['codeception', 'phpspec'];
+
+    private const TIMEOUT = 180.0; // 3 minutes
+
+    private $composerExecutableFinder;
+
+    public function __construct(ComposerExecutableFinder $composerExecutableFinder)
+    {
+        $this->composerExecutableFinder = $composerExecutableFinder;
+    }
+
+    public function install(string $adapterName): void
+    {
+        Assert::oneOf($adapterName, self::OFFICIAL_ADAPTERS);
+
+        $adapterComposerPackageName = sprintf('infection/%s-adapter', $adapterName);
```

Comment:
wouldn't it be clearer to have a `OFFICIAL_ADAPTER_MAP` instead where the values of `OFFICIAL_ADAPTERS ` are the keys and their associated values the packagist package name?